src/main/scala/centaur/api/CromwellClient.scala

```diff
@@ -1,25 +1,104 @@
+  // Akka HTTP needs both the actor system and a materializer
+  final implicit val system = ActorSystem("centaur-acting-like-a-system")
+  final implicit val materializer: ActorMaterializer = ActorMaterializer(ActorMaterializerSettings(system))
+
+  def submit(workflow: Workflow): Try[SubmittedWorkflow] = {
+    val params = Map("wdlSource" -> Option(workflow.data.wdl),
+      "workflowInputs" -> workflow.data.inputs,
+      "workflowOptions" -> insertSecrets(workflow.data.options)
+    ) collect { case (name, Some(value)) => (name, value) }
+    val formData = FormData(params).toEntity
+
+    val submittedWorkflow = for {
+      response <- Http().singleRequest(HttpRequest(HttpMethods.POST, CentaurConfig.cromwellUrl + "/api/workflows/v1", entity=formData))
+      entity <- response.toEntity.to[CromwellStatus]
+    } yield SubmittedWorkflow(UUID.fromString(entity.id), CentaurConfig.cromwellUrl, workflow)
+    sendReceiveFutureCompletion(submittedWorkflow)
+  }
+
+  def status(workflow: SubmittedWorkflow): Try[WorkflowStatus] = {
+    val workflowStatus = for {
+      response <- Http().singleRequest(HttpRequest(uri = CentaurConfig.cromwellUrl + "/api/workflows/v1/" + workflow.id + "/status"))
+      entity <- response.toEntity.to[CromwellStatus]
+    } yield WorkflowStatus(entity.status)
+
+    sendReceiveFutureCompletion(workflowStatus)
+  }
+
+  def metadata(workflow: SubmittedWorkflow): Try[WorkflowMetadata] = {
+    val workflowMetadata = for {
+      response <- Http().singleRequest(HttpRequest(uri = CentaurConfig.cromwellUrl + "/api/workflows/v1/" + workflow.id + "/metadata"))
+      entity <- response.toEntity.to[String]
+    } yield WorkflowMetadata.fromMetadataJson(entity).toOption.get
+
+    sendReceiveFutureCompletion(workflowMetadata)
+  }
+
+  lazy val backends: Try[CromwellBackends] = {
+    val backends = for {
+      response <- Http().singleRequest(HttpRequest(uri = CentaurConfig.cromwellUrl + "/api/workflows/v1/backends"))
+      entity <- response.toEntity.to[CromwellBackends]
+    } yield entity
+
+    sendReceiveFutureCompletion(backends)
+  }
 
   /**
     * Ensure that the Future completes within the specified timeout. If it does not, or if the Future fails,
     * will return a Failure, otherwise a Success
     */
   def awaitFutureCompletion[T](x: Future[T], timeout: FiniteDuration) = Try(Await.result(x, timeout))
   def sendReceiveFutureCompletion[T](x: Future[T]) = awaitFutureCompletion(x, CentaurConfig.sendReceiveTimeout)
+
+  private def insertSecrets(options: Option[String]): Option[String] = {
+    import DefaultJsonProtocol._
+    val tokenKey = "refresh_token"
+
+    def addToken(optionsMap: Map[String, JsValue]): Map[String, JsValue] = {
+      CentaurConfig.optionalToken match {
+        case Some(token) if optionsMap.get(tokenKey).isDefined => optionsMap + (tokenKey -> JsString(token))
+        case _ => optionsMap
+      }
+    }
+
+    options match {
```

Comment:
blargh, I almost monkeyed around w/ this function when i moved it across and backed out figuring someone would bark at me for doing something unrelated (granted @kshakir was tagged as a reviewer and that's his patented move so I might still have been right!) - I can't win ;)